Code/Engine/RendererFoundation/Device/Implementation/Device.cpp

```diff
@@ -642,6 +642,64 @@ ezGALTextureHandle ezGALDevice::CreateTexture(
   return ezGALTextureHandle();
 }
 
+ezResult ezGALDevice::ReplaceExisitingNativeObject(ezGALTextureHandle hTexture, void* pExisitingNativeObject)
+{
+  ezGALTexture* pTexture = nullptr;
+  if (m_Textures.TryGetValue(hTexture, pTexture))
+  {
+    for (auto it = pTexture->m_ResourceViews.GetIterator(); it.IsValid(); ++it)
+    {
+      ezGALResourceView* pResourceView = nullptr;
+
+      if (m_ResourceViews.TryGetValue(it.Value(), pResourceView))
+      {
+        EZ_VERIFY(pResourceView->DeInitPlatform(this).Succeeded(), "DeInitPlatform should never fail.");
+      }
+    }
+    for (auto it = pTexture->m_RenderTargetViews.GetIterator(); it.IsValid(); ++it)
+    {
+      ezGALRenderTargetView* pRenderTargetView = nullptr;
+
+      if (m_RenderTargetViews.TryGetValue(it.Value(), pRenderTargetView))
+      {
+        EZ_VERIFY(pRenderTargetView->DeInitPlatform(this).Succeeded(), "DeInitPlatform should never fail.");
+      }
+    }
+    for (auto it = pTexture->m_UnorderedAccessViews.GetIterator(); it.IsValid(); ++it)
+    {
+      ezGALUnorderedAccessView* pUnorderedAccessView = nullptr;
+
+      if (m_UnorderedAccessViews.TryGetValue(it.Value(), pUnorderedAccessView))
+      {
+        EZ_VERIFY(pUnorderedAccessView->DeInitPlatform(this).Succeeded(), "DeInitPlatform should never fail.");
+      }
+    }
+
+    EZ_VERIFY(pTexture->DeInitPlatform(this).Succeeded(), "DeInitPlatform should never fail.");
+    EZ_VERIFY(pTexture->ReplaceExisitingNativeObject(pExisitingNativeObject).Succeeded(),
+      "Failed to replace native texture, make sure the input is valid.");
+    EZ_VERIFY(pTexture->InitPlatform(this, {}).Succeeded(),
+      "InitPlatform failed on a texture the previously succeded in the same call, is the new native object valid?");
+    
+    for (auto it = pTexture->m_RenderTargetViews.GetIterator(); it.IsValid(); ++it)
```

Comment:
You should of course also re-init resourceViews and UnorderedAccessViews here ;)

Reply:
And I was about to click the submit button! Thanks for looking it over. Will fix it today.